addons/binding/org.openhab.binding.nest/ESH-INF/thing/thermostat.xml

```diff
@@ -0,0 +1,136 @@
+			</channel>
+			<channel id="max_set_point" typeId="setpoint">
+				<label>Max set point</label>
+				<description>Max set point in celcius</description>
+			</channel>
+			<channel id="min_set_point" typeId="setpoint">
+				<label>Min set point</label>
+				<description>Min set point in celcius</description>
+			</channel>
+			<channel id="can_heat" typeId="readonlyswitch">
+				<label>Can Heat</label>
+				<description>If the system can heat.</description>
+			</channel>
+			<channel id="can_cool" typeId="readonlyswitch">
+				<label>Can Cool</label>
+				<description>If the system can cool.</description>
+			</channel>
+			<channel id="has_fan" typeId="readonlyswitch">
+				<label>Has Fan</label>
+				<description>If the system has a fan.</description>
+			</channel>
+			<channel id="has_leaf" typeId="readonlyswitch">
+				<label>Has Leaf</label>
+				<description>If the system is currently in a leaf mode.</description>
+			</channel>
+			<channel id="sunlight_correction_enabled" typeId="readonlyswitch">
+				<label>Sunlight Correction Enabled</label>
+				<description>If sunlight correction is enabled.</description>
+			</channel>
+			<channel id="sunlight_correction_active" typeId="readonlyswitch">
+				<label>Sunlight correction Active</label>
+				<description>If sunlight correction is active.</description>
+			</channel>
+			<channel id="using_emergency_heat" typeId="readonlyswitch">
+				<label>Using Emergency Heat</label>
+				<description>If the system is currently using emergency heat.</description>
+			</channel>
+			<channel id="locked" typeId="readonlyswitch">
+				<label>Locked</label>
+				<description>If the system currently has a locked heat/cool range</description>
+			</channel>
+			<channel id="locked_max_set_point" typeId="setpoint">
+				<label>Locked max set point</label>
+				<description>Locked max set point in celcius</description>
+			</channel>
+			<channel id="locked_min_set_point" typeId="setpoint">
+				<label>Locked Min set point</label>
+				<description>Locked min set point in celcius</description>
+			</channel>
+			<channel id="time_to_target_mins" typeId="timeToTarget">
+				<label>Time to Targer</label>
+				<description>Time to target temperature in mins (approximately).</description>
+			</channel>
+		</channels>
+
+		<properties>
+			<property name="vendor">Nest</property>
+		</properties>
+	</thing-type>
+
+	<channel-type id="temperature">
+		<item-type>Number</item-type>
+		<label>Temperature</label>
+		<description>Current temperature in degrees Celsius</description>
+		<category>Temperature</category>
+		<state readOnly="true" />
+	</channel-type>
+
+	<channel-type id="setpoint">
+		<item-type>Number</item-type>
+		<label>Temperature</label>
+		<description>Set point in degrees Celsius</description>
+		<category>Temperature</category>
+		<state readOnly="false" min="18" max="42" />
+	</channel-type>
+
+	<channel-type id="readonlyswitch">
```

Comment:
same comment as before. Please create individual types and move your label/descriptions from the channels to the types.